source/MaterialXCore/Element.cpp

```diff
@@ -325,6 +327,23 @@ ConstElementPtr Element::getRoot() const
     return root;
 }
 
+const string& Element::getActiveColorSpace(bool returnClosest) const
```

Comment:
Currently elements from an imported document if this method is will used to find an elements target colorspace this will just be that elements color space if there is one and hence no transform will occur. 
There may be a higher level element which would be the local target. If the document's root is used but is empty then no transform will occur. This allows for the search for the target to find the higher level element to be used as the source which could include the colorspace set on the children of the root of the imported document.

Reply:
I can see where you might want to transform a color to the "next colorspace up" in the hierarchy, but it's not clear to me why you would select the "furthest but non-root" colorspace that this function computes.  I would imagine that would only handle cases where exactly one level of libraries was imported into a document, rather than handling the general, hierarchical case.

As a side note, I think it's reasonable for now to handle just the common case of a single "working" colorspace, with all source colors being transformed into this single space.  For the future, I agree with your goal of handling complex colorspace hierarchies, but I believe this will require additional research and design work.

Reply:
I was thinking about this and was leaning towards leaving this out. I'll revert this and the import colorspace changes as I'd like to work out the bigger picture of the fact that we'd like to differentiate between "import" which embeds one file into another and a "reference" which the dependent document keeps it's own properties relative to a local "root". This traversal is partially a "guess" to try and go as far up as possible to find this local "root".